vpc.tf

```diff
@@ -0,0 +1,130 @@
+resource "aws_vpc" "fsb_vpc" {
+  cidr_block = "173.152.0.0/16"
+}
+
+resource "aws_subnet" "public_subnet_1" {
+  vpc_id     = aws_vpc.fsb_vpc.id
+  cidr_block = "173.152.1.0/24"
+  availability_zone = "us-east-1a"
+  map_public_ip_on_launch = true
+}
+
+resource "aws_subnet" "public_subnet_2" {
+  vpc_id     = aws_vpc.fsb_vpc.id
+  cidr_block = "173.152.2.0/24"
+  availability_zone = "us-east-1b"
+  map_public_ip_on_launch = true
+}
+
+resource "aws_subnet" "private_subnet_1" {
+  vpc_id     = aws_vpc.fsb_vpc.id
+  cidr_block = "173.152.3.0/24"
+  availability_zone = "us-east-1a"
+}
+
+resource "aws_subnet" "private_subnet_2" {
+  vpc_id     = aws_vpc.fsb_vpc.id
+  cidr_block = "173.152.4.0/24"
+  availability_zone = "us-east-1b"
+}
+
+resource "aws_internet_gateway" "fsb_igw" {
+  vpc_id = aws_vpc.fsb_vpc.id
+}
+
+# Create a public route table
+resource "aws_route_table" "my_public_route_table" {
+  vpc_id = aws_vpc.fsb_vpc.id
+
+  # Route to the Internet Gateway
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.fsb_igw.id
+  }
+}
+
+# Associate the public subnet with the public route table
+resource "aws_route_table_association" "my_public_route_table_association1" {
+  subnet_id      = aws_subnet.public_subnet_1.id
+  route_table_id = aws_route_table.my_public_route_table.id
+}
+
+resource "aws_route_table_association" "my_public_route_table_association2" {
+  subnet_id      = aws_subnet.public_subnet_2.id
+  route_table_id = aws_route_table.my_public_route_table.id
+}
+
+# Create 2 NAT Gateways
+resource "aws_nat_gateway" "nat_gateway_1" {
+  allocation_id = aws_eip.nat_gateway_1.id
+  subnet_id     = aws_subnet.private_subnet_1.id
+}
+
+resource "aws_nat_gateway" "nat_gateway_2" {
+  allocation_id = aws_eip.nat_gateway_2.id
+  subnet_id     = aws_subnet.private_subnet_2.id
```

Comment:
could you please associate nat gateway with public subnet as it is use to connect private instance to the internet